src/simple_cache.erl

```diff
@@ -30,88 +30,118 @@
 %%% Public API.
 -export([init/1]).
 -export([get/4, get/5]).
--export([flush/1, flush/2, clear/2, create_value/4]).
+-export([flush/1, flush/2, flush/3, clear/2, create_value/4]).
 
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 %% Public API.
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 %% @doc Initializes a cache.
 -spec init(string()) -> ok.
 init(CacheName) ->
-  CacheName = ets:new(CacheName, [
-    named_table, {read_concurrency, true}, public, {write_concurrency, true}
-  ]),
-  ok.
+    CacheName = ets:new(CacheName, [
+        named_table, {read_concurrency, true}, public, {write_concurrency, true}
+    ]),
+    ok.
+
+%% @doc Deletes all keys in the given cache.
+-spec flush(string()) -> true.
+flush(CacheName) ->
+    true = ets:delete_all_objects(CacheName).
 
 %% @doc Deletes the keys that match the given ets:matchspec() from the cache.
 -spec flush(string(), term()) -> true.
 flush(CacheName, Key) ->
-  ets:delete(CacheName, Key).
+    ets:delete(CacheName, Key).
+
+% for safe flushing of custom caching layer on top of mnesia
+-spec flush(string(), term(), atom()) -> true.
+flush(CacheName, {get_data_by_key, MnesiaKey, _Type} = Key, TableName) ->
+    FlushFun =
+        fun() ->
+            mnesia:lock({TableName, MnesiaKey}, write),
+            ets:delete(CacheName, Key)
+        end,
+    {atomic, Result} = db_functions:apply_transaction(FlushFun),
```

Comment:
If you're doing it this way, it's no longer like a library.

Either think of a way to make this generic to keep it like a library, or just move simple_cache to our code as a set of modules and make changes there.